pkg/controller/kibanaassociation/labels.go

```diff
@@ -21,24 +20,24 @@ const (
 	AssociationLabelNamespace = "kibanaassociation.k8s.elastic.co/namespace"
 )
 
-// NewResourceSelector selects resources labeled as related to the named association.
-func NewResourceSelector(name string) client.MatchingLabels {
-	return client.MatchingLabels(map[string]string{
-		AssociationLabelName: name,
-	})
-}
-
-func hasBeenCreatedBy(object metav1.Object, kibana *kbv1.Kibana) bool {
-	labels := object.GetLabels()
-	if name, ok := labels[AssociationLabelName]; !ok || name != kibana.Name {
+var createdByKibana association.CreatedBy = func(created, creator metav1.Object) bool {
+	labels := created.GetLabels()
+	if name, ok := labels[AssociationLabelName]; !ok || name != creator.GetName() {
 		return false
 	}
-	if ns, ok := labels[AssociationLabelNamespace]; !ok || ns != kibana.Namespace {
+	if ns, ok := labels[AssociationLabelNamespace]; !ok || ns != creator.GetNamespace() {
 		return false
 	}
 	return true
 }
```

Comment:
Since it's basically the same function for Kibana, APMServer and EnterpriseSearch, could we make a single generic one?

```go
// -- in common/association

// ServesAssociationPurpose returns true if the given resource has been created to serve an association
// based on the given labels.
func ServesAssociationPurpose(resource metav1.Object, associationLabels map[string]string) bool {
    return maps.IsSubset(associationLabels, resource.GetLabels())
}

// -- in kibanaassociation

// KibanaAssociationSelector returns labels to set on resources created for association purposes.
func KibanaAssociationSelector(kb kbv1.Kibana) map[string]string{
    return map[string]string{
        AssociationLabelName: kb.Name,
        AssociationLabelNamespace: kb.Namespace,
    }
}

association.DeleteOrphanedResources(ctx, c, kibana, NewResourceSelector(kibana.Name), KibanaAssociationSelector(kb))
```

The difference between `NewResourceSelector` and `KibanaAssociationSelector` is not very clear to me though, I must be missing something.